### ISSUE TITLE
Unable to enroll after enrolling and unenrolling once

### ISSUE THREAD

**Issue by VaibhavMisra** [msintuneappsdk/ms-intune-app-sdk-ios#67]
Our LOB app already uses ADAL for authentication. Hence, as per integration docs Intune for SDK, I am configuring Intune with our authoridy, ClientID and redirectID. The the app enroll user using 'IntuneMAMEnrollmentManager.instance().registerAndEnrollAccount(user)' on login and unenroll using 'IntuneMAMEnrollmentManager.instance().deRegisterAndUnenrollAccount(user, withWipe: false)' on logout.

This works well the first time, and I can see the policy taking effect. However, when I logout once and try to login again, I get following error when the app tries enroll again: 

StatusCode: 203
Error String: "The operation failed because the SDK could not access the user's AAD token.  The application should prompt the user for credentials to refresh the user's AAD token."

Is there something I'm missing here?

**Comment by patlewis-MSFT** [comment]
Which version of ADAL are you using with your LOB app?

**Comment by VaibhavMisra** [comment]
> Which version of ADAL are you using with your LOB app?

ADAL version is 2.7.13. 

Interestingly, if I close the app and run it again, its starts working and enrollment happens successfully. I am testing this in debug mode right now, will it have any effect?

**Comment by VaibhavMisra** [comment]
Hi! 

Is there any update? We are stuck due to this.

PS: I also upgraded ADAL to 4.0.5 and iOS deployment target to 11.0, still facing the same issue.

**Comment by VaibhavMisra** [comment]
Team, this is an urgent issue where our release is blocked due to this issue. Could you provide a solution or even a workaround?

**Comment by idcuesta** [comment]
Hey there,

I have seen some similar issue when started integrating Intune SDK into our project. The problem in my case was that I assumed `[IntuneMAMPolicyManager instance]` method was a singleton, so I initialised it (aadClientIdOverride, aadRedirectUriOverride, etc...) in a dispatch_once. After unenroll was called, instance object changed (new one was instantiated) and this configuration got lost, so when we tried to enrol after authenticating in ADAL again, it was not using proper values and was not sharing session with ADAL. This was before they introduced `IntuneMAMSettings`.

Hope that helps !
